services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/EventController.java

```diff
@@ -0,0 +1,1023 @@
+            List<BsmMessageCountProgressionEvent> list = new ArrayList<>();
+            list.add(MockEventGenerator.getBsmMessageCountProgressionEvent());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = bsmMessageCountProgressionEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(bsmMessageCountProgressionEventRepo.find(query));
+        }
+    }
+
+    @RequestMapping(value = "/events/bsm_message_count_progression/count", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("hasRole('USER') || hasRole('ADMIN')")
+    public ResponseEntity<Long> countBsmMessageCountProgressionEvents(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "full_count", required = false, defaultValue = "true") boolean fullCount,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            return ResponseEntity.ok(1L);
+        } else {
+            Query query = bsmMessageCountProgressionEventRepo.getQuery(intersectionID, startTime, endTime, false);
+
+            long count;
+
+            if (fullCount) {
+                count = bsmMessageCountProgressionEventRepo.getQueryFullCount(query);
+            } else {
+                count = bsmMessageCountProgressionEventRepo.getQueryResultCount(query);
+            }
+
+            log.debug("Found: {} BsmMessageCountProgressionEvents", count);
+            return ResponseEntity.ok(count);
+        }
+    }
+
+    @RequestMapping(value = "/events/bsm_events", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID) and @PermissionService.hasRole('USER')) ")
+    public ResponseEntity<List<BsmEvent>> findBsmEvents(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "latest", required = false, defaultValue = "false") boolean latest,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            List<BsmEvent> list = new ArrayList<>();
+            list.add(MockEventGenerator.getBsmEvent());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = bsmEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(bsmEventRepo.find(query));
```

Comment:
issue(blocking): executing query second time for logging purposes

Reply:
Extra query has been removed